### ISSUE TITLE
XdsNameResolver blindly propagates XdsClient errors

### ISSUE THREAD

**Issue by ejona86** [grpc/grpc-java#8950]
If a watch in XdsClient fails, XdsNameResolver will propagate the error directly to the channel without adding context and without setting a proper status.

https://github.com/grpc/grpc-java/blob/3b9ff362b9b64b3615c493a4f94eed6d00f711e5/xds/src/main/java/io/grpc/xds/XdsNameResolver.java#L725

A user saw
```
 [Channel<2441>: (xds:///<snip>)] Failed to resolve name. status=Status{code=NOT_FOUND, description=Requested entity was not found., cause=null}
```

And accompanying RPC failures:
```
Caused by: java.lang.RuntimeException: io.grpc.StatusRuntimeException with message: NOT_FOUND: Requested entity was not found.
```

That RPC failure _is very bad_, because it isn't clear it is from XdsNR and should be UNAVAILABLE.

**Comment by YifeiZhuang** [comment]
Hi @ejona86 , curious did you encounter the error? Shall we log the real error and always throw "UNAVAILABLE" to the client, while copying the cause and description details?

**Comment by ejona86** [comment]
At the least we should wrap and repackage the error. For example: "UNAVAILABLE: Unable to load Route 'Foo'. xDS server returned: NOT_FOUND: Requested entity was not found." This is unrelated to logging.

**Comment by ejona86** [comment]
This is part of the (now-available) v1.45.0 release.
